### ISSUE TITLE
Suggestion - Missing optuna in the requirements file. 

### ISSUE THREAD

**Issue by RaulSokolova** [mikazlopes/training-farm#8]
Would be possible to please add the specific version to the requirement packages ? 

Im having issue with some package. What version do you have for flask and eventlet ?

**Comment by mikazlopes** [comment]
Hi, 

I am still cleaning up my requirements; as for the flask and eventless versions, here is a full list of everything I have installed. Several modules are part of the new features I am implementing.

Package                    Version
-------------------------- ------------
absl-py                    2.0.0
aiodns                     3.1.0
aiohttp                    3.8.2
aiohttp-cors               0.7.0
aiosignal                  1.3.1
ale-py                     0.8.1
alembic                    1.13.0
alpaca-trade-api           3.0.2
antlr4-python3-runtime     4.12.0
appdirs                    1.4.4
appnope                    0.1.3
asttokens                  2.4.0
async-timeout              4.0.3
attrs                      23.1.0
autocommand                2.2.2
AutoROM                    0.6.1
AutoROM.accept-rom-license 0.6.1
backcall                   0.2.0
backports.csv              1.0.7
beautifulsoup4             4.12.2
bidict                     0.22.1
bitarray                   2.9.0
bitstring                  4.1.4
blessed                    1.20.0
blinker                    1.6.3
blis                       0.7.11
bokeh                      3.3.3
bottle                     0.12.25
box2d-py                   2.3.5
cachetools                 5.3.1
catalogue                  2.0.10
ccxt                       3.1.60
certifi                    2023.7.22
cffi                       1.16.0
charset-normalizer         2.1.1
cheroot                    10.0.0
CherryPy                   18.9.0
click                      8.1.7
cloudpathlib               0.16.0
cloudpickle                2.2.1
colorful                   0.5.5
colorlog                   6.8.0
confection                 0.1.4
contourpy                  1.1.1
cryptography               41.0.4
cvxpy                      1.3.2
cycler                     0.12.1
cymem                      2.0.8
dask                       2023.12.1
decorator                  5.1.1
deprecation                2.1.0
distlib                    0.3.7
distributed                2023.12.1
dnspython                  2.4.2
ecos                       2.0.12
elegantrl                  0.3.6
empyrical                  0.5.5
en-core-web-sm             3.7.1
eventlet                   0.33.3
exceptiongroup             1.1.3
exchange-calendars         4.5
executing                  2.0.0
Farama-Notifications       0.0.4
feedparser                 6.0.11
filelock                   3.12.4
finrl                      0.3.6
Flask                      3.0.0
Flask-SocketIO             5.3.6
fonttools                  4.43.1
frozendict                 2.3.8
frozenlist                 1.4.0
fsspec                     2023.9.2
future                     0.18.3
google-api-core            2.12.0
google-auth                2.23.2
google-auth-oauthlib       1.0.0
googleapis-common-protos   1.60.0
gpustat                    1.1.1
greenlet                   3.0.0
grpcio                     1.59.0
gym                        0.26.2
gym-notices                0.0.8
gymnasium                  0.29.1
h11                        0.14.0
holidays                   0.39
html5lib                   1.1
huggingface-hub            0.20.2
idna                       3.4
importlib-metadata         7.0.1
importlib-resources        6.1.0
inflect                    7.0.0
iniconfig                  2.0.0
ipython                    8.16.1
itsdangerous               2.1.2
jaraco.collections         5.0.0
jaraco.context             4.3.0
jaraco.functools           4.0.0
jaraco.text                3.12.0
jedi                       0.19.1
Jinja2                     3.1.2
joblib                     1.3.2
jqdatasdk                  1.9.1
jsonschema                 4.19.1
jsonschema-specifications  2023.7.1
kazoo                      2.9.0
kiwisolver                 1.4.5
korean-lunar-calendar      0.3.1
langcodes                  3.3.0
locket                     1.0.0
lxml                       4.9.3
lz4                        4.3.3
Mako                       1.3.0
Markdown                   3.5
markdown-it-py             3.0.0
MarkupSafe                 2.1.3
matplotlib                 3.8.0
matplotlib-inline          0.1.6
mdurl                      0.1.2
more-itertools             10.1.0
mpmath                     1.3.0
msgpack                    1.0.3
multidict                  5.2.0
multitasking               0.0.11
murmurhash                 1.0.10
mysql-connector-python     8.2.0
mysqlclient                2.2.1
networkx                   3.1
nltk                       3.8.1
numpy                      1.26.0
nvidia-ml-py               12.535.108
oauthlib                   3.2.2
opencensus                 0.11.3
opencensus-context         0.1.3
opencv-python              4.8.1.78
optuna                     3.5.0
optuna-dashboard           0.14.0
osqp                       0.6.3
overrides                  7.4.0
packaging                  23.2
pandas                     2.1.1
pandas-datareader          0.10.0
parso                      0.8.3
partd                      1.4.1
pdfminer.six               20231228
peewee                     3.16.3
pexpect                    4.8.0
pickleshare                0.7.5
Pillow                     10.0.1
pip                        23.2.1
platformdirs               3.11.0
pluggy                     1.3.0
ply                        3.11
portend                    3.2.0
preshed                    3.0.9
prometheus-client          0.17.1
prompt-toolkit             3.0.39
protobuf                   4.21.12
psutil                     5.9.5
psycopg2-binary            2.9.9
ptyprocess                 0.7.0
pure-eval                  0.2.2
py-spy                     0.3.14
pyarrow                    14.0.1
pyarrow-hotfix             0.6
pyasn1                     0.5.0
pyasn1-modules             0.3.0
pycares                    4.4.0
pycparser                  2.21
pydantic                   1.10.13
pyfolio                    0.9.2
pygame                     2.1.0
Pygments                   2.16.1
pyluach                    2.2.0
PyMySQL                    1.1.0
pyparsing                  3.1.1
pyportfolioopt             1.5.5
pytest                     7.4.2
pytest-mock                3.11.1
python-dateutil            2.8.2
python-docx                1.1.0
python-engineio            4.7.1
python-socketio            5.9.0
pytz                       2023.3.post1
pyutils                    0.0.13
PyYAML                     6.0
qdldl                      0.1.7.post0
ray                        2.7.0
referencing                0.30.2
regex                      2023.12.25
requests                   2.31.0
requests-oauthlib          1.3.1
rich                       13.6.0
rpds-py                    0.10.4
rsa                        4.9
safetensors                0.4.1
scikit-learn               1.3.1
scipy                      1.11.3
scs                        3.2.3
seaborn                    0.13.0
setuptools                 68.0.0
sgmllib3k                  1.0.0
Shimmy                     1.1.0
simple-websocket           1.0.0
six                        1.16.0
smart-open                 6.4.0
sortedcontainers           2.4.0
soupsieve                  2.5
spacy                      3.7.2
spacy-legacy               3.0.12
spacy-loggers              1.0.5
SQLAlchemy                 1.4.49
srsly                      2.4.8
stable-baselines3          2.1.0
stack-data                 0.6.3
stockstats                 0.5.4
swig                       4.1.1
sympy                      1.12
tblib                      3.0.0
tempora                    5.5.0
tensorboard                2.14.1
tensorboard-data-server    0.7.1
tensorboardX               2.6.2.2
textblob                   0.17.1
thinc                      8.2.2
threadpoolctl              3.2.0
thriftpy2                  0.4.17
tokenizers                 0.15.0
tomli                      2.0.1
toolz                      0.12.0
torch                      2.1.0
torch-tb-profiler          0.4.3
tornado                    6.4
tqdm                       4.66.1
traitlets                  5.11.2
transformers               4.36.2
typer                      0.9.0
typing_extensions          4.8.0
tzdata                     2023.3
urllib3                    1.26.17
utils                      1.0.1
vaderSentiment             3.3.2
virtualenv                 20.21.0
wasabi                     1.1.2
wcwidth                    0.2.8
weasel                     0.3.4
webencodings               0.5.1
websocket-client           1.6.4
websockets                 10.4
Werkzeug                   3.0.0
wheel                      0.41.2
wrds                       3.1.6
wsproto                    1.2.0
xyzservices                2023.10.1
yarl                       1.9.2
yfinance                   0.2.31
zc.lockfile                3.0.post1
zict                       3.0.0
zipp                       3.17.0

**Comment by RaulSokolova** [comment]
Thank you so much...
I've found the bug with the latest version of evenlet==0.34.3 , not sure why, but it does require to import trio first :) 

```
import trio
import eventlet
eventlet.monkey_patch()
```

Here is the bug link in case someone might need it.
https://github.com/eventlet/eventlet/issues/805

**Comment by mikazlopes** [comment]
Thanks! When you send the PR with the new file name, feel free to also add this to main.py.
